portable/GCC/ARM_CM4_MPU/portmacro.h

```diff
@@ -71,6 +71,14 @@ typedef unsigned long    UBaseType_t;
  * not need to be guarded with a critical section. */
     #define portTICK_TYPE_IS_ATOMIC    1
 #endif
+
+/* Cortex-M7 r0p0 and r0p1 cores require a workaround for ARM errata 837070.
+ * When using this port on a Cortex-M7 r0p0 or r0p1 core, define
+ * configTARGET_ARM_CM7_r0p0 or configTARGET_ARM_CM7_r0p1 to 1 in your
+ * FreeRTOSConfig.h. */
+#if ( ( configTARGET_ARM_CM7_r0p0 == 1 ) || ( configTARGET_ARM_CM7_r0p1 == 1 ) )

```

Comment:
Consider having the user define configENABLE_ERRATA_837070_WORKAROUND directly, rather than deriving the constant (with the "port" prefix) from the constants on line 79.  It would be simpler, and asserts make sure it is only defined on the r0p0 and r0p1 cores.  It would also prevent having to provide defaults for the constants on line 79.  If that is changed then the comment telling the user what to do if the assert fails needs updating too.

Reply:
Done.